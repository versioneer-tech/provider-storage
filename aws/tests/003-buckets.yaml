---
apiVersion: pkg.internal/v1beta1
kind: Storage
metadata:
  name: s-jane
spec:
  bucketAccessRequests:
  - bucketName: s-john
    reason: Need WriteOnly access
    requestedAt: "2025-09-29T10:20:00Z"
  buckets: []
  principal: jane
status:
  conditions:
  - lastTransitionTime: "2024-01-01T00:00:00Z"
    message: 'Unready resources: access-key-jane, observe-jane-restricted-s-john,
      and user-jane'
    reason: Creating
    status: "False"
    type: Ready
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: AccessKey
metadata:
  annotations:
    crossplane.io/composition-resource-name: access-key-jane
  labels:
    crossplane.io/composite: s-jane
  name: jane
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-jane
    uid: ""
spec:
  forProvider:
    userRef:
      name: jane
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
  writeConnectionSecretToRef:
    name: jane
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: observe-jane-restricted-s-john
  labels:
    crossplane.io/composite: s-jane
  name: observe-jane-restricted-s-john
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-jane
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: iam.aws.m.upbound.io/v1beta1
      kind: Policy
      metadata:
        name: jane.restricted.s-john
        namespace: default
  managementPolicies:
  - Observe
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: User
metadata:
  annotations:
    crossplane.io/composition-resource-name: user-jane
  labels:
    crossplane.io/composite: s-jane
  name: jane
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-jane
    uid: ""
spec:
  forProvider: {}
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
