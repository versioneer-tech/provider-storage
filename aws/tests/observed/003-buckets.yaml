---
apiVersion: s3.aws.upbound.io/v1beta2
kind: Bucket
metadata:
  annotations:
    crossplane.io/claim-name: alice-003
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: bucket-alice-003-alice-003
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: s3.aws.upbound.io/v1beta2
kind: Bucket
metadata:
  annotations:
    crossplane.io/claim-name: alice-003
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: bucket-alice-003-alice-shared-003
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: Policy
metadata:
  annotations:
    crossplane.io/claim-name: alice-003
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: policy-alice-003.owner.alice-003
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: Policy
metadata:
  annotations:
    crossplane.io/claim-name: alice-003
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: policy-alice-003.owner.alice-shared-003
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: User
metadata:
  annotations:
    crossplane.io/claim-name: alice-003
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: user-alice-003
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: UserPolicyAttachment
metadata:
  annotations:
    crossplane.io/claim-name: alice-003
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: user-policy-attachement-alice-003.owner.alice-003
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: UserPolicyAttachment
metadata:
  annotations:
    crossplane.io/claim-name: alice-003
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: user-policy-attachement-alice-003.owner.alice-shared-003
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: AccessKey
metadata:
  annotations:
    crossplane.io/claim-name: alice-003
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: access-key-alice-003
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
