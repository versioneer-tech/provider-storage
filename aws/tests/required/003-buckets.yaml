---
apiVersion: v1
data:
  random-string: abcde-fghij
kind: ConfigMap
metadata:
  name: random-string
  namespace: crossplane-system
