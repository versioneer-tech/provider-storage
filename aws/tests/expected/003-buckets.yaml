---
apiVersion: pkg.internal/v1beta1
kind: XStorage
metadata:
  labels:
    crossplane.io/claim-name: alice-003
    crossplane.io/claim-namespace: default
  name: alice-003
spec:
  bucketAccessRequests:
  - bucketName: bob-shared-003
    permission: ReadWrite
  buckets:
  - bucketName: alice-003
    discoverable: false
  - bucketName: alice-shared-003
    discoverable: true
  owner: alice-003
status:
  conditions:
  - lastTransitionTime: "2024-01-01T00:00:00Z"
    message: 'Unready resources: check-policy-alice-003.readwrite.bob-shared-003'
    reason: Creating
    status: "False"
    type: Ready
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: AccessKey
metadata:
  annotations:
    crossplane.io/composition-resource-name: access-key-alice-003
  generateName: alice-003-
  labels:
    crossplane.io/composite: alice-003
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-003
    uid: ""
spec:
  forProvider:
    userRef:
      name: alice-003
  providerConfigRef:
    name: storage-aws
  writeConnectionSecretToRef:
    name: alice-003
    namespace: default
---
apiVersion: s3.aws.upbound.io/v1beta2
kind: Bucket
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-alice-003-alice-003
    xstorages.pkg.internal/discoverable: "false"
  generateName: alice-003-
  labels:
    crossplane.io/composite: alice-003
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-003
    uid: ""
spec:
  forProvider:
    region: eu-central-1
  providerConfigRef:
    name: storage-aws
---
apiVersion: s3.aws.upbound.io/v1beta2
kind: Bucket
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-alice-003-alice-shared-003
    xstorages.pkg.internal/discoverable: "true"
  generateName: alice-003-
  labels:
    crossplane.io/composite: alice-003
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-003
    uid: ""
spec:
  forProvider:
    region: eu-central-1
  providerConfigRef:
    name: storage-aws
---
apiVersion: kubernetes.crossplane.io/v1alpha2
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: check-policy-alice-003.readwrite.bob-shared-003
  generateName: alice-003-
  labels:
    crossplane.io/composite: alice-003
  name: policy-observer-alice-003.readwrite.bob-shared-003
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-003
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: iam.aws.upbound.io/v1beta1
      kind: Policy
      metadata:
        name: alice-003.readwrite.bob-shared-003
  managementPolicies:
  - Observe
  providerConfigRef:
    name: storage-kubernetes
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: Policy
metadata:
  annotations:
    crossplane.io/composition-resource-name: policy-alice-003.owner.alice-003
  generateName: alice-003-
  labels:
    crossplane.io/composite: alice-003
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-003
    uid: ""
spec:
  forProvider:
    policy: |
      {
        "Version": "2012-10-17",
        "Statement": [
          {
            "Effect": "Allow",
            "Action": [ "s3:*" ],
            "Resource": [
              "arn:aws:s3:::<no value>",
              "arn:aws:s3:::<no value>/*"
            ]
          }
        ]
      }
  providerConfigRef:
    name: storage-aws
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: Policy
metadata:
  annotations:
    crossplane.io/composition-resource-name: policy-alice-003.owner.alice-shared-003
  generateName: alice-003-
  labels:
    crossplane.io/composite: alice-003
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-003
    uid: ""
spec:
  forProvider:
    policy: |
      {
        "Version": "2012-10-17",
        "Statement": [
          {
            "Effect": "Allow",
            "Action": [ "s3:*" ],
            "Resource": [
              "arn:aws:s3:::<no value>",
              "arn:aws:s3:::<no value>/*"
            ]
          }
        ]
      }
  providerConfigRef:
    name: storage-aws
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: User
metadata:
  annotations:
    crossplane.io/composition-resource-name: user-alice-003
  generateName: alice-003-
  labels:
    crossplane.io/composite: alice-003
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-003
    uid: ""
spec:
  forProvider: {}
  providerConfigRef:
    name: storage-aws
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: UserPolicyAttachment
metadata:
  annotations:
    crossplane.io/composition-resource-name: user-policy-attachement-alice-003.owner.alice-003
  generateName: alice-003-
  labels:
    crossplane.io/composite: alice-003
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-003
    uid: ""
spec:
  forProvider:
    policyArnRef:
      name: alice-003.owner.alice-003
    userRef:
      name: alice-003
  providerConfigRef:
    name: storage-aws
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: UserPolicyAttachment
metadata:
  annotations:
    crossplane.io/composition-resource-name: user-policy-attachement-alice-003.owner.alice-shared-003
  generateName: alice-003-
  labels:
    crossplane.io/composite: alice-003
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XStorage
    name: alice-003
    uid: ""
spec:
  forProvider:
    policyArnRef:
      name: alice-003.owner.alice-shared-003
    userRef:
      name: alice-003
  providerConfigRef:
    name: storage-aws
